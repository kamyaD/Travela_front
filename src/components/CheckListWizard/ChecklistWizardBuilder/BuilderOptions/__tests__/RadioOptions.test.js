import React from 'react';
import { shallow } from 'enzyme';
import RadioOption from '../RadioOptions';

const props = {
  item: {
    id: 'kldskd',
    name: 'are you serious?',
    behaviour: {
      type: 'NOTIFY_EMAIL',
      payload: 'john@example.com'
    }
  },
  updateBehaviour: jest.fn(),
  deleteQuestion: jest.fn(),
};

describe('<RadioOption />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<RadioOption {...props} />);
    expect(wrapper.find('div'));
  });

  it('should handle update behaviour', () => {
    
    const wrapper = mount(<RadioOption {...props} />);
    const mockEvents = { target: { value: 'Passport?'} };
    const btn = wrapper.find('#option-name-input');
    btn.simulate('change', mockEvents);
    expect(wrapper.updateBehaviour).toBeCalled;
  });

  it('should handle delete icon onChange', () => {
    const wrapper = mount(<RadioOption {...props} />);
    const mockEvents = { target: { value: 'Passport?'} };
    const deleteIcon = wrapper.find('button#option-del-icon');
    deleteIcon.simulate('click', mockEvents);
    expect(wrapper.deleteQuestion).toBeCalled;
  });

  it('should handle show behaviour btn', () => {
    
    const wrapper = mount(<RadioOption {...props} />);
    const mockEvents = jest.fn();
    const btn = wrapper.find('.set-behaviour-btn');
    btn.simulate('click', mockEvents);
    expect(wrapper.showListOfBehaviours).toBeCalled;
  });

  it('should handle show behaviour btn', () => {
    
    const wrapper = mount(<RadioOption {...props} />);
    wrapper.find('.set-behaviour-btn').first().simulate('click');
    const input = wrapper.find('#emailToSend');
    input.first().simulate('change');
    expect(input).toHaveLength(1);
    expect(wrapper.instance().props.updateBehaviour).toBeCalled;
  });
});
