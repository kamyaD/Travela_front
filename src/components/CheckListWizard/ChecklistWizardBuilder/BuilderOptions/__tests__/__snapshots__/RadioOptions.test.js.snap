// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioOption /> shoudl call deleteRadioOptionBehaviour when button is clicked 1`] = `
<div>
  <div>
    <p>
      Enter Option 1
    </p>
    <div
      className="prompt-item"
    >
      <input
        className="prompt-input"
        id="option-name-input"
        onChange={[Function]}
        placeholder="Yes"
        type="text"
        value="yes"
      />
      <DeleteIcon
        id="option-del-icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="behaviour-container"
    >
      <div
        className="set-behaviour-btn-container"
      />
      <div>
        <div>
          <div
            className="remove-set-behaviour"
          >
             
            <span>
              <ContextMenu
                classNames="table__menu-container"
              >
                <MenuItem
                  classNames="delete"
                  onClick={[Function]}
                >
                  Delete behaviour
                </MenuItem>
              </ContextMenu>
            </span>
            Upon selecting Option 1, enable
          </div>
          <Behaviours
            optionId=""
            order={1}
            type=""
            updateBehaviour={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<RadioOption /> should render RenderCheckbox component properly 1`] = `
<div>
  <div>
    <p>
      Enter Option 1
    </p>
    <div
      className="prompt-item"
    >
      <input
        className="prompt-input"
        id="option-name-input"
        onChange={[Function]}
        placeholder="Yes"
        type="text"
        value="yes"
      />
      <DeleteIcon
        id="option-del-icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="behaviour-container"
    >
      <div
        className="set-behaviour-btn-container"
      >
        <button
          className="set-behaviour-btn"
          onClick={[Function]}
          type="button"
        >
          Set behaviour
        </button>
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`<RadioOption /> should render RenderCheckbox component properly 2`] = `
<div>
  <div>
    <p>
      Enter Option 1
    </p>
    <div
      className="prompt-item"
    >
      <input
        className="prompt-input"
        id="option-name-input"
        onChange={[Function]}
        placeholder="Yes"
        type="text"
        value="yes"
      />
      <button
        className="builder-delete-icon"
        id="option-del-icon"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="delete-icon"
          src="delete-icon.svg"
        />
      </button>
    </div>
    <div
      className="behaviour-container"
    >
      <div
        className="set-behaviour-btn-container"
      >
        <button
          className="set-behaviour-btn"
          onClick={[Function]}
          type="button"
        >
          Set behaviour
        </button>
      </div>
      <div />
    </div>
  </div>
</div>
`;
